### PR TITLE
add optional gamepad support

### PR DESCRIPTION
**Pull request by ryan-haskell**
Hey there @w0rm ,

I recently started playing around with the [Gamepad API](https://developer.mozilla.org/en-US/docs/Web/API/Gamepad_API) and Elm yesterday, and was able to add optional controller support to Mogee!

I've been playing Mogee on my couch with an Xbox 360 controller!

Thanks for the cool game- hope you have a controller to try it out with (`a` to jump, left joystick to move)

Thanks again! 😄 
Ryan

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -33,6 +33,23 @@
...
+        window.requestAnimationFrame(listenToGamepad)
+      }
+    }
+    var gamepadConnected = false
+    window.addEventListener('gamepadconnected', function(e) {  
+      gamepadConnected = true  
+      window.requestAnimationFrame(listenToGamepad)
+    })
+    window.addEventListener('gamepaddisconnected', function(e) {
+      gamepad = false
```
Comment (w0rm) [inline index.html]: Should this rather set gamepadConnected to false?

**Comment by w0rm** [comment]
Hey @ryannhg this looks cool! Unfortunately I don’t have any gamepad at my disposal. Are there any caveats in supporting different controllers? I’ve heard about this package  https://github.com/xarvh/elm-gamepad by @xarvh

File: index.html (hunk 1 above)
Comment (ryan-haskell) [reply index.html]: yep 😅 i should clean that up

**Comment by xarvh** [comment]
As long as you use only two axis and a button you /may/ get away with it.
You may want to test different browsers, different browsers versions on different operative systems, since the axis/buttons reported by the browsers tend to change in number and layout very inconsistently.
I'll try to run a few tests when I have time.

**Comment by ryan-haskell** [comment]
The implementation I did was so simple, it just looks for a single button and left and right on the joystick. I haven't added in full support for more controls, just thought jump+move would be fun.

It worked with my Steam controller, but used the up to jump instead of the `a` button.

**Comment by w0rm** [comment]
@xarvh cool, thanks for taking a look! I can also merge this PR and then you can try from https://unsoundscapes.itch.io/mogee without having to compile. Would this be helpful?

**Comment by xarvh** [comment]
@w0rm yeah, would be less work for me ^_^

**Comment by w0rm** [comment]
@ryannhg thanks for your contribution!

**Comment by xarvh** [comment]
Did a quick test with latest Chrome on Ubuntu 1904 and Windows 10, two different xbox 360 knockoffs, the game itself seems to work well, but I can't open or navigate the menu.

Also, the digital pad is not working, which might be a problem with digital-only pads such as SNES USB replicas.

Under Windows, false contacts might cause a zombie gamepad to appear as element 0 of the pseudo-array, while the actual gamepad takes position 1, depending on how portable you want to be you might want to [filter that out](https://github.com/xarvh/elm-gamepad/blob/master/port/gamepadPort.js#L46)

@ryannhg if you are interested in the Gamepad API, you can check the docs of the [Elm Gamepad package](https://package.elm-lang.org/packages/xarvh/elm-gamepad/latest/).

**Pull request merged**